docs/content/integrations/airlift/operator-migration/kubernetes-pod-operator.mdx

```diff
@@ -0,0 +1,159 @@
+# Operator migration guides: Migrating usage of KubernetesPodOperator
+
+In this page, we'll explain migrating an Airflow `KubernetesPodOperator` to Dagster.
+
+### Background
+
+The KubernetesPodOperator in Apache Airflow enables users to execute containerized tasks within Kubernetes pods as part of their data pipelines.
+
+```python file=/integrations/airlift/operator_migration/kubernetes_pod_operator.py
+from airflow.providers.cncf.kubernetes.operators.pod import KubernetesPodOperator
+
+k8s_hello_world = KubernetesPodOperator(
+    task_id="hello_world_task",
+    name="hello-world-pod",
+    image="bash:latest",
+    cmds=["bash", "-cx"],
+    arguments=['echo "Hello World!"'],
+)
+```
+
+### Dagster equivalent
+
+The Dagster equivalent is to use the <PyObject object="PipesK8sClient" module="dagster_k8s"/> to execute a task within a Kubernetes pod.
+
+```python file=/integrations/airlift/operator_migration/using_k8s_pipes.py
+from dagster_k8s import PipesK8sClient
+
+from dagster import AssetExecutionContext, asset
+
+container_cfg = {
+    "name": "hello-world-pod",
+    "image": "bash:latest",
+    "command": ["bash", "-cx"],
+    "args": ['echo "Hello World!"'],
+}
+
+
+@asset
+def execute_hello_world_task(context: AssetExecutionContext):
+    return (
+        PipesK8sClient()
+        .run(
+            context=context.op_execution_context,
+            base_pod_meta={"name": "hello-world-pod"},
+            base_pod_spec={"containers": [container_cfg]},
+        )
+        .get_results()
+    )
+```
+
+### Migrating the operator
+
+Migrating the operator breaks down into a few steps:
+
+1. Ensure that your Dagster deployment has access to the Kubernetes cluster.
+2. Write an <PyObject object="asset" module="dagster"/> that executes the task within a Kubernetes pod using the <PyObject object="PipesK8sClient" module="dagster_k8s"/>.
+3. Use `dagster-airlift` to proxy execution of the original task to Dagster.
+
+### Step 1: Ensure access to the Kubernetes cluster
+
+First, you need to ensure that your Dagster deployment has access to the Kubernetes cluster where you want to run your tasks. The <PyObject object="PipesK8sClient" module="dagster_k8s"/> accepts `kubeconfig` and `kubecontext`, and `env` arguments to configure the Kubernetes client.
+
+Here's an example of what this might look like when configuring the client to access an EKS cluster:
+
+```python file=/integrations/airlift/operator_migration/k8s_eks_fake_example.py startafter=start_client endbefore=end_client
+from dagster_k8s import PipesK8sClient
+
+eks_client = PipesK8sClient(
+    # The client will have automatic access to all
+    # environment variables in the execution context.
+    env={**AWS_CREDENTIALS, "AWS_REGION": "us-west-2"},
+    kubeconfig_file="path/to/kubeconfig",
+    kube_context="my-eks-cluster",
+)
```

Comment:
[1] Shouldn't this also be in the first example?

Reply:
I don't think so - this is designed to just be the equivalent to the above airflow code, which doesn't reference any environment variables. I want to just introduce this part when it's necessary.